### PR TITLE
More old code cleanup. Review by @VladUreche or @heathermiller.

### PR DESCRIPTION
**Pull request by vigdorchik**

### CODE REVIEW

**Comment by vigdorchik** [comment]
PLS REBUILD ALL,

File: src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala [hunk 1]
```diff
@@ -124,15 +124,11 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
     def group = comment flatMap (_.group) getOrElse "No Group"
     override def inTemplate = inTpl
     override def toRoot: List[MemberImpl] = this :: inTpl.toRoot
-    def inDefinitionTemplates = this match {
```
Comment (VladUreche) [inline src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala:127]: Here you're missing the usecase special treatment. Say you have:

```
trait T { def foo(implicit i: Int): Int }
class C extends T { 
  /** @usecase def foo: Int */
  def foo(implicit i: Int): Int = 3 
}
```

The inDefinitionTemplates for the usecase must be the same as the ones for the member, but this is not true for usecase symbols -- so we have to special case that.

Comment (vigdorchik) [inline src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala:127]: @VladUreche I thought usecases generate 'NonTemplateMemberImpl' where I override inDefinitionTemplates. Is it not the case?

**Comment by VladUreche** [comment]
LGTM, you were right, sorry!

**Comment by gkossakowski** [comment]
@vigdorchik: Improve the commit message please:
- "More code cleanup" is totally not meaningful message. Drop "More" and add "Scaladoc" in front of "code cleanup"
- Put reviews as a separate line in a commit message

**Pull request merged**